### ISSUE TITLE
[Bug]: `@babel/register` has incorrect `sourceMaps` default settings

### ISSUE THREAD

**Issue by Domiii** [babel/babel#14021]
### 💻

- [ ] Would you like to work on a fix?

### How are you using Babel?

@babel/register or @babel/node

### Input code

Any code.

### Configuration file name

babel.config.json

### Configuration

```json
{
  sourceMaps: undefined
}
```

### Current and expected behavior

**Current Behavior**

`@babel/register` in [this line](https://github.com/babel/babel/blob/master/packages/babel-register/src/node.js#L64) sets the default to `"both"`:

`sourceMaps: opts.sourceMaps === undefined ? "both" : opts.sourceMaps`

**Expected Behavior**

[According to the documentation](https://babeljs.io/docs/en/options#sourcemaps), the default should be `false`.

### Environment

@babel/register: 7.10.1

### Possible solution

Change code to:

```
sourceMaps: opts.sourceMaps === undefined ? false : opts.sourceMaps
```

**Comment by nicolo-ribaudo** [comment]
We should update the `@babel/register` [docs](https://babeljs.io/docs/en/babel-register) to say that it overwrites the default value. The reason we enable them by default is you don't have access to the compiled sources, so the only way to make sense of errors is via source maps.

**Comment by Domiii** [comment]
@nicolo-ribaudo Right, i came across the same problem. `source-maps` are a must. Agreed that it is a documentation issue.

**Comment by Domiii** [comment]
@nicolo-ribaudo I would recommend, also adding such divergence (`register` being different from the rest) to [the configuration docs](https://babeljs.io/docs/en/options#sourcemaps), since most people would not consider the idea that configuration is suddenly divergent (because divergence happens so rarely).
